groth16/src/snark.rs

```diff
@@ -0,0 +1,36 @@
+use algebraic::errors::Result;
+use franklin_crypto::bellman::pairing::ff::PrimeField;
```

Comment:
We don't have to import franklin_crypto, as its an implement of `plonk`. 

For here, we  can use the pairing crate instead `franklin` one.